.github/workflows/publish-commit-bottles.yml

```
@@ -335,11 +336,23 @@ jobs:
 
           echo "head_sha=$(git rev-parse HEAD)" >> "$GITHUB_OUTPUT"
 
-      - name: generate build provenance
-        uses: github-early-access/generate-build-provenance@main
+      - name: Generate build provenance
+        # GitHub Attestations are still in early public access, and we've seen
+        # some sporadic errors when generating attestations.
+        # Rather than fail outright, we allow this step to fail and fall
+        # through to a last-ditch retry below. Longer term, we should put this
+        # above the 'pr-pull' step, which should be split into separate 'pull'
+        # and 'push to GitHub Packages' phases.
+        continue-on-error: true
+        uses: actions/attest-build-provenance@v1
         with:
           subject-path: '${{steps.pr-pull.outputs.bottle_path}}/*.tar.gz'
-  
+
+      - name: Generate build provenance (last ditch)
```

Something like this might work in case the action is not idempotent:
```suggestion
      - name: Generate build provenance (last ditch)
        if: always() && steps.first-attestation-attempt.conclusion == 'failure'
```
but the first step will need
```
id: first-attestation-attempt
```